File: suppet-backend/src/main/java/com/dyplom/suppet/service/puppetdb/status/PuppetDbStatusService.java
package com.dyplom.suppet.service.puppetdb.status;

import com.dyplom.suppet.service.puppetdb.AbstractPuppetDbService;
import org.springframework.stereotype.Service;

@Service
public class PuppetDbStatusService extends AbstractPuppetDbService {

    PuppetDbStatusService() {
        super("PuppetDbStatusService");
    }

    @Override
    protected String getBaseUrl() {
        return "https://puppet-db.home:8081/status/v1/services/";
    }

    @Override
    protected String getBaseEndpointName() {
        return "puppetdb-status";
    }
}

File: suppet-frontend/src/app/commons/puppet-db/status/puppet-db-status.service.ts
import { Injectable } from '@angular/core';
import {AbstractPuppetDbService} from "../abstract-puppet-db.service";
import {HttpClient} from "@angular/common/http";

@Injectable({
  providedIn: 'root'
})
export class PuppetDbStatusService extends AbstractPuppetDbService {

  constructor(http: HttpClient) {
    super(http);
  }

  protected getBaseUrl(): string {
    return "/puppet-db/status";
  }
}
